includes/OptionsHandler.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * WordPress Options-table Session Handler
+ *
+ * Like the original WP Session Handler, this class uses the WordPress
+ * Options table for data storage.
+ *
+ * @package WP Session Manager
+ * @subpackage Handlers
+ * @since 2.0
+ */
+namespace EAMann\Sessionz\Handlers;
+
+use EAMann\Sessionz\Handler;
+use EAMann\Sessionz\Objects\Option;
+
+/**
+ * Use an associative array to store session data so we can cut down on
+ * round trips to an external storage mechanism (or just leverage an in-
+ * memory cache for read performance).
+ */
+class OptionsHandler implements Handler {
+
+	/**
+	 * Sanitize a potential Session ID so we aren't fetching broken data
+	 * from the options table.
+	 *
+	 * @param string $id
+	 *
+	 * @return string
+	 */
+	private function sanitize( $id ) {
+		return preg_replace( "/[^A-Za-z0-9_]/", '', $id );
+	}
+
+	/**
+	 * Helper function to either add or update an existing option in the table.
+	 *
+	 * @param string $id
+	 * @param mixed  $data
+	 */
+	private function add_or_update( $id, $data ) {
+		if ( ! add_option( $id, $data, '', 'no' ) ) {
+			update_option( $id, $data );
+		}
+	}
+
+	/**
+	 * Get an item out of a WordPress option
+	 *
+	 * @param string $id
+	 *
+	 * @return bool|string
+	 */
+	protected function _read( $id ) {
+		$session_id = $this->sanitize( $id );
+
+		$data = get_option( "_wp_session_$session_id" );
+		$expires = intval( get_option( "_wp_session_expires_$session_id" ) );
+		if ( false !== $data ) {
+			/** @var Option $item */
+			$item = new Option( $data, $expires );
+			if ( ! $item->is_valid() ) {
+				delete_option( "_wp_session_$session_id" );
+				delete_option( "_wp_session_expires_$session_id" );
+				return false;
+			}
+
+			return $session;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Purge an item from the database immediately.
+	 *
+	 * @param string   $id
+	 * @param callable $next
+	 *
+	 * @return mixed
+	 */
+	public function delete( $id, $next ) {
+		$session_id = $this->sanitize( $id );
+
+		delete_option( "_wp_session_$session_id" );
+		delete_option( "_wp_session_expires_$session_id" );
+		return $next( $id );
+	}
+
+	/**
+	 * Update the Options table by removing any items that are no longer valid.
+	 *
+	 * @param int      $maxlifetime
+	 * @param callable $next
+	 *
+	 * @global \wpdb $wpdb
+	 *
+	 * @return mixed
+	 */
+	public function clean( $maxlifetime, $next ) {
+		global $wpdb;
+
+		// Session is expired if now - item.time > maxlifetime
+		// Said another way, if  item.time < now - maxlifetime
+		$filter = intval( time() - $maxlifetime );
+		$keys = $wpdb->get_results( "SELECT option_name, option_value FROM $wpdb->options WHERE option_name LIKE '_wp_session_expires_%' AND option_value > $filter ORDER BY option_value LIMIT 0, 1000" );
```

Comment:
Even though you're sanitizing the `$filter` variable just above this, my spidey-sense tingles whenever I don't see SQL queries going through `$wpdb->prepare()`.